nixos/hardware/vultr/bare-metal.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, inputs, modulesPath, config, ... }:
+{
+  imports = [
+    ../../mixins/cloud-init.nix
+    (modulesPath + "/profiles/all-hardware.nix")
+  ];
+
+  boot.kernelParams = [ "vultr" ]; # need for cloud-init on baremetal server
+
+  # Fixes boot issue
+  boot.initrd.services.swraid.enable = true;
+
+  services.cloud-init = {
+    config = ''
+      system_info:
+        distro: nixos
+        network:
+          renderers: [networkd]
+      users:
+         - root
+
+      disable_root: false
+      preserve_hostname: false
+
+      datasource:
+        Vultr:
+          url: http://169.254.169.254
+          retries: 30
+          timeout: 10
+          wait: 5
+
+      cloud_init_modules:
+       - set_hostname
+       - update_hostname
+       - update_etc_hosts
+      cloud_config_modules: []
+      cloud_final_modules: []
+    '';
+  };
+  systemd.services.cloud-init.enable = false;
```

Comment:
Can you elaborate with cloud-init.service is disabled here? Does undo the configuration above?

Reply:
@nitrotm do you remember why it was set that way?

Reply:
right, that was because the cloud-init local setup was working (and enough to set hostname) but the rest was failing.

Reply:
something about the Vultr datasource not being able to setup more advanced network parameters if I recall properly; I am not sure if that's still actual.

Reply:
thanks, I will try to reproduce the issue then